backend/tests/tasks_test.py

```diff
@@ -337,24 +336,187 @@ def test_pumps_slow_dispense(microlab):
+    """
+    If 'time' is omitted, rate defaults to maxSpeed; behavior should match the 'within-speed' branch,
+    yielding exactly one dispense_time and one final None.
+    """
+    # Omit 'time' → volume=5, so rate = maxSpeed = 5.0
+    monkeypatch.setattr(
+        microlab,
+        "getPumpSpeedLimits",
+        lambda pump_name: {"minSpeed": 1.0, "maxSpeed": 5.0}
+    )
+    monkeypatch.setattr(
+        microlab,
+        "pumpDispense",
+        lambda pump_name, volume, duration: 1.11
+    )
+    monkeypatch.setattr(
+        microlab,
+        "uptime_seconds",
+        lambda: 0.0
+    )
+
+    fn = tasks.pump(microlab, {"pump": "X", "volume": 5})
+
+    # 1) First yield: dispense_time at max speed
+    dispense_time = next(fn)
+    assert dispense_time == pytest.approx(1.11)
+
+    # 2) Second yield: single `None` (completion)
+    res = next(fn)
+    assert res is None
+
+    with pytest.raises(StopIteration):
+        next(fn)
+
+
+@pytest.mark.microlab_data({"reactor-reagent-dispenser": {"minSpeed": 2.0, "maxSpeed": 5.0}})
+def test_pumps_burst_mode_with_partial(monkeypatch, microlab) -> None:
+    """
+    If desired rate is below minSpeed, the function should:
+      1. Dispense in repeated 1-second bursts at minSpeed,
+      2. Yield (interval - exec_time) for each full burst (with exec_time=0 here),
+      3. After exhausting full bursts, yield a final fractional delay for the remaining volume,
+      4. Then yield exactly one None (completion).
+    """
+    # volume=5, time=4 => rate=1.25 < minSpeed=2.0
+    monkeypatch.setattr(
+        microlab,
+        "getPumpSpeedLimits",
+        lambda pump_name: {"minSpeed": 2.0, "maxSpeed": 5.0}
+    )
+    # Simulate instant execution of pumpDispense so exec_time = 0
+    monkeypatch.setattr(
+        microlab,
+        "uptime_seconds",
+        lambda: 100.0
+    )
+    monkeypatch.setattr(
+        microlab,
+        "pumpDispense",
+        lambda pump_name, volume, duration=None: None
+    )
+
+    fn = tasks.pump(microlab, {"pump": "X", "volume": 5, "time": 4})
+
+    # Compute expected interval = (minSpeed / rate) - 1 = (2.0 / 1.25) - 1 = 1.6 - 1 = 0.6
+    expected_interval = (2.0 / 1.25) - 1.0  # 0.6
+
+    # Two full bursts: dispenses 2 mL twice → 4 mL total → yields two intervals
+    for _ in range(2):
+        res = next(fn)
+        assert res == pytest.approx(expected_interval, rel=1e-3)
+
+    # Now 4 mL has been dispensed; remaining = 1 mL
+    # Final fractional wait = remaining / rate = 1 / 1.25 = 0.8
+    final_fractional = next(fn)
+    assert final_fractional == pytest.approx(0.8, rel=1e-3)
+
+    # Next: single None (completion)
+    res = next(fn)
+    assert res is None
 
+    # Generator should now be exhausted
+    with pytest.raises(StopIteration):
+        next(fn)
 
-@pytest.mark.skip(reason="temporary bypass to restore GitHub build workflow")
+
+# MAINTAIN PID
+
+# @pytest.mark.skip(reason="temporary bypass to restore GitHub build workflow")
```

Comment:
It looks like these are passing, so we can remove this comment and the one below?